05/ContactForm.js

```diff
@@ -1,10 +1,109 @@
-import React from 'react';
+import React, { useReducer, useState } from 'react';
 
-import account from './account';
+function ContactForm() {
+    const reducer = (state, { name, value }) => {
+        const newState = { ...state };
+        newState[name] = value;
+        return newState;
+    };
 
-const ContactForm = () => {
-    console.log(account);
-    return <form>??</form>;
-};
+    const initialState = { names: '', email: '', message: '', subject: '' };
+    const [state, dispatch] = useReducer(reducer, initialState);
+    const { names, email, message, subject } = state;
+    const [errors, setErrors] = useState([]);
+    const [success, setSuccess] = useState(false);
+
+    const handleSubmit = (e) => {
+        e.preventDefault();
+
+        const errorsList = [];
+        setErrors([]);
+
+        if (names.length === 0) {
+            errorsList.push('Name is required');
+        }
+
+        if (!email.includes('@')) {
+            errorsList.push('Email is required');
+        }
+
+        if (message.length === 0) {
+            errorsList.push('Message is required');
+        }
+
+        if (subject.length === 0) {
+            errorsList.push('Subject is required');
+        }
+
+        if (errorsList.length > 0) {
+            setErrors(errorsList);
+        } else {
+            setSuccess(true);
+        }
```

Comment:
Idealnie by było gdybyś zrobił to bardziej uniwersalnie tj. tak:
https://devmentor.pl/b/walidacja-formularza-w-javascript